pkg/actuators/machine/actuator.go

```diff
@@ -538,6 +538,24 @@ func (a *Actuator) getMachineInstances(cluster *clusterv1.Cluster, machine *mach
 		return nil, err
 	}
 
+	status := &providerconfigv1.AWSMachineProviderStatus{}
+	err = a.codec.DecodeProviderStatus(machine.Status.ProviderStatus, status)
+
```

Comment:
don't we want to error here? seems legit even if we want to fallback when no error.

Reply:
We could I guess, but my intention was to not introduce any new behavior if any part of the new flow here fails. Previously, this would have happily returned an instance even with no ProviderStatus. Seems better for now to keep it that way.